Tasks/CondaAuthenticateV0/condaauthenticatemain.ts

```diff
@@ -0,0 +1,319 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as fs from 'fs';
+import * as constants from './constants';
+import * as util from 'azure-pipelines-tasks-packaging-common/util';
+import * as pkgLocationUtils from 'azure-pipelines-tasks-packaging-common/locationUtilities';
+import * as url from 'url';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        let configCondarc = tl.getInput(constants.CondaAuthenticateTaskInput.ConfigFile);
+        if (!configCondarc) {
+            throw new Error(tl.loc('ConfigCondarcDoesNotExist', configCondarc));
+        } else if (!configCondarc.endsWith('.condarc')) {
+            throw new Error(tl.loc('CondarcNotCondarc', configCondarc));
+        } else {
+            console.log(tl.loc("AuthenticatingThisConfigCondarc", configCondarc));
+        }
+
+        const condarcFile = fs.readFileSync(configCondarc, 'utf8');
+        const condarcContents = condarcFile.split(/\r?\n/);
+
+        const channelPieces = {};  //Entries:  <full channel str as listed> : [<half after '@' symbol>, <TOKEN_ENV_VAR>]
+
+        let channelsIndex = -1;
+        let channelsAlias = '';
+        let defaultChannelsIndex = -1;
+        let customChannelsIndex = -1;
+        let customMultichannelsIndex = -1;
+        let restoreFreeChannel = false;
+        for (let i = 0; i < condarcContents.length; i++) {
+            const line = condarcContents[i].trim();
+            if (line.startsWith('channels:')) {
+                channelsIndex = i;
+            } else if (line.startsWith('channels_alias: ')) {
+                const aliases = line.split(/\s+/);
+                if (aliases.length !== 2) {
+                    // Throw error for invalid format
+                    throw new Error(tl.loc("InvalidCondarcFormat", 'channel_alias'));
+                }
+                channelsAlias = aliases[1].replace(/['"]+/g, '')
+                getAliasPieces(channelsAlias);
+            } else if (line.startsWith('default_channels:')) {
```

Comment:
Are these channels case sensitive for conda client? E.g. would Default_channels work for conda client